controllers/azuremachine_controller.go

```diff
@@ -309,12 +309,6 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
 		machineScope.SetFailureMessage(errors.Errorf("Azure VM state is %s", machineScope.VMState()))
 		conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMRunningCondition, infrav1.VMProvisionFailedReason, clusterv1.ConditionSeverityWarning, "")
 		machineScope.SetNotReady()
-		// If VM failed provisioning, delete it so it can be recreated
-		err := ams.DeleteVM(ctx)
```

Comment:
the DeleteVM function can also be removed, it shouldn't be used anywhere else

Reply:
Thanks for the review! Are you talking about [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/controllers/azuremachine_reconciler.go#L125) function?

Reply:
yes, that one

Reply:
Thank you, done.